src/TokensTracker/Trackers/TrackerBase.cs

```diff
@@ -0,0 +1,146 @@
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.Plugins.Trackers
+{
+    abstract class TrackerBase
+    {
+        protected bool _shouldTrackHistory;
+        protected uint _maxResults;
+        protected IStore _db;
+        private ISnapshot _levelDbSnapshot;
+        protected NeoSystem _neoSystem;
+
+        protected TrackerBase(IStore db, uint maxResult, bool shouldTrackHistory, NeoSystem neoSystem)
+        {
+            _db = db;
+            _maxResults = maxResult;
+            _shouldTrackHistory = shouldTrackHistory;
+            _neoSystem = neoSystem;
+        }
+
+        public abstract void OnPersist(NeoSystem system, Block block, DataCache snapshot, IReadOnlyList<Blockchain.ApplicationExecuted> applicationExecutedList);
+
+        public void ResetBatch()
+        {
+            _levelDbSnapshot?.Dispose();
+            _levelDbSnapshot = _db.GetSnapshot();
+        }
+
+        public void Commit()
+        {
+            _levelDbSnapshot?.Commit();
+        }
+
+        public IEnumerable<(TKey key, TValue val)> QueryTransfers<TKey, TValue>(byte dbPrefix, UInt160 userScriptHash, ulong startTime, ulong endTime)
+                where TKey : ISerializable, new()
+                where TValue : class, ISerializable, new()
+        {
+            var prefix = new[] { dbPrefix }.Concat(userScriptHash.ToArray()).ToArray();
+            byte[] startTimeBytes, endTimeBytes;
+            if (BitConverter.IsLittleEndian)
+            {
+                startTimeBytes = BitConverter.GetBytes(BinaryPrimitives.ReverseEndianness(startTime));
+                endTimeBytes = BitConverter.GetBytes(BinaryPrimitives.ReverseEndianness(endTime));
+            }
+            else
+            {
+                startTimeBytes = BitConverter.GetBytes(startTime);
+                endTimeBytes = BitConverter.GetBytes(endTime);
+            }
+            var transferPairs = _db.FindRange<TKey, TValue>(prefix.Concat(startTimeBytes).ToArray(), prefix.Concat(endTimeBytes).ToArray());
+            return transferPairs;
+        }
+
+        protected static byte[] Key(byte prefix, ISerializable key)
+        {
+            byte[] buffer = new byte[key.Size + 1];
+            using (MemoryStream ms = new MemoryStream(buffer, true))
+            using (BinaryWriter writer = new BinaryWriter(ms))
+            {
+                writer.Write(prefix);
+                key.Serialize(writer);
+            }
+            return buffer;
+        }
+
+        protected void Put(byte prefix, ISerializable key, ISerializable value)
+        {
+            _levelDbSnapshot.Put(Key(prefix, key), value.ToArray());
+        }
+
+        protected void Delete(byte prefix, ISerializable key)
+        {
+            _levelDbSnapshot.Delete(Key(prefix, key));
+        }
+
+        protected TransferRecord GetTransferRecord(UInt160 asset, Array stateItems)
```

Comment:
Why do you use the NEP-11 `TransferRecord` in the base class?

Reply:
It's a common  transfer record, moved it to base class.